_posts/2018-09-13-API-Gateway-Types-Compared.md

```diff
@@ -0,0 +1,153 @@
+
+Because a picture speaks a thousand words, here is a visualisation:
+![Architecture](/assets/posts/2018-09-13-API-Gateway-Types-Compared/architecture.png)
+
+# The original API Gateway: Edge Optimized
+At its release in July 2015 (read the [announcement](https://aws.amazon.com/blogs/aws/amazon-api-gateway-build-and-run-scalable-application-backends/)), API Gateway allowed access to Lambda and publicly available HTTP endpoints. At that time, Lambdas could not be placed in VPCs - this feature was [released](https://aws.amazon.com/blogs/aws/new-access-resources-in-a-vpc-from-your-lambda-functions/) in February 2016. In its initial version, API Gateway came paired with a CloudFront distribution. The combined API Gateway and CloudFront would later be called **Edge Optimized** API Gateway, refering to the Edge locations available in CloudFront.
+
+![Edge Optimized](/assets/posts/2018-09-13-API-Gateway-Types-Compared/edge-optimized-1.png)
+
+Important details to remember regarding the first version of the Edge Optimized API Gateway:
+* It uses CloudFront distributions, but you *can't* edit the distribution. Adding a WAF, for example, is not possible.
+* It can access other sources, but those sources have to be publicly accessible.
+
+# Regional API Gateways
+In November 2017, Amazon [introduced](https://aws.amazon.com/about-aws/whats-new/2017/11/amazon-api-gateway-supports-regional-api-endpoints/) the **Regional** API Gateway. This variant, as its name implies, is deployed and accessible in a single region. In other words: it's the Edge Optimized API Gateway without the CloudFront distribution.
+
+![Regional](/assets/posts/2018-09-13-API-Gateway-Types-Compared/regional.png)
+
+Benefits to this variant are:
+* You can deploy your own CloudFront distribution in front of your API Gateway. This distribution *can* have a WAF.
+* If your API Gateway is going to be accessed from the same AWS region, the Regional API Gateway will have less latency.
+
+# Integration with Private VPCs
+A big downside to API Gateway has alweays been that any HTTP backend behind the API Gateway needed to be publicly accessible. And because the public IP address of the API Gateway is unknown or unpredictable, IP whitelisting at the backend system was not a viable option. To solve this issue, Amazon [introduced](https://aws.amazon.com/about-aws/whats-new/2017/11/amazon-api-gateway-supports-endpoint-integrations-with-private-vpcs/) integration with private VPCs.
```

Comment:
`alweays` -> `always`

Reply:
`Amazon [introduced]` add when so it matches your other introduction references

Reply:
Good point, fixed